### PR TITLE
[16.0][IMP] fs_storage: pattern matching should only apply to file name, not full path

### PR DESCRIPTION
**Pull request by tuantrantg**

### CODE REVIEW

File: fs_storage/models/fs_storage.py [hunk 1]
```diff
@@ -408,7 +408,8 @@ def find_files(self, pattern, relative_path="", **kw) -> list[str]:
             return []
         regex = re.compile(pattern)
         for file_path in self.fs.ls(relative_path, detail=False):
-            if regex.match(file_path):
+            # fs.ls returns a relative path
```
Comment (simahawk) [inline fs_storage/models/fs_storage.py:411]: As this method is deprecated, does it makes sense to improve it? Unless, this is a fix...

**Review by simahawk — COMMENTED** [review]
A remark for the commit msg
```
[IMP or FIX] fs_storage: improve/fix find_files pattern match

bla bla explanation 
```

File: fs_storage/models/fs_storage.py (hunk 1 above)
Comment (tuantrantg) [reply fs_storage/models/fs_storage.py:411]: yes, it's a fix

**Review by lmignon — APPROVED** [review]

**Pull request merged**